### PR TITLE
dkimproxy: init at 1.4.1

### PR DESCRIPTION
**Pull request by Ekleog**
###### Motivation for this change

OpenSMTPD not supporting OpenDKIM but only dkimproxy (for the time being), dkimproxy may prove useful.

Note: I have just tested the executable appears to build at the moment, not yet that it works, not having yet a fully functional OpenSMTPD setup. The package builds on NixOS with sandboxing, and all executables appear to run without obvious errors. I'll come back after having finished setting up an OpenSMTPD using it if that's required for merging.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/mail/dkimproxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  buildInputs = [ perl ];
+  propagatedBuildInputs = [ Error MailDKIM MIMEtools NetServer ];
+
+  meta = with stdenv.lib; {
+    description = "SMTP-proxy that signs and/or verifies emails";
+    homepage = http://dkimproxy.sourceforge.net/;
+    license = licenses.gpl2Plus;
+    maintainers = [ ];
```
Comment (Mic92) [inline pkgs/servers/mail/dkimproxy/default.nix]: Do you want to maintain this package?

Comment (Ekleog) [inline pkgs/servers/mail/dkimproxy/default.nix]: Oh, I'd have thought `buildPerlPackages` did more than just adding `perl` in `buildInputs`, thanks for the simplification!

As for maintaining, I can maintain it. Pushing a commit that adds me as maintainer right now. :)

**Comment by Ekleog** [comment]
It turns out `buildPerlPackage` was actually required: without it, `dkim_responder.pl` fails with the following error message:
```
Can't locate MIME/Entity.pm in @INC (you may need to install the MIME::Entity module) (@INC contains: /nix/store/vapgb6j9jydhklv1zas4a9syii6mwbf5-dkimproxy-1.4.1/lib /nix/store/vapgb6j9jydhklv1zas4a9syii6mwbf5-dkimproxy-1.4. /home/ekleog/.nix-profile/lib/perl5/site_perl /nix/var/nix/profiles/default/lib/perl5/site_perl /run/current-system/sw/lib/perl5/site_perl/5.24.1/x86_64-linux-thread-multi /run/current-system/sw/lib/perl5/site_perl/5.24.1 /run/current-system/sw/lib/perl5/site_perl /nix/store/fd7m7yxql9nx05arfcqnwfh7vdwcgvxb-perl-5.24.1/lib/perl5/site_perl/5.24.1/x86_64-linux-thread-multi /nix/store/fd7m7yxql9nx05arfcqnwfh7vdwcgvxb-perl-5.24.1/lib/perl5/site_perl/5.24.1 /nix/store/fd7m7yxql9nx05arfcqnwfh7vdwcgvxb-perl-5.24.1/lib/perl5/5.24.1/x86_64-linux-thread-multi /nix/store/fd7m7yxql9nx05arfcqnwfh7vdwcgvxb-perl-5.24.1/lib/perl5/5.24.1 .) at /home/ekleog/.nix-profile/bin/dkim_responder.pl line 26.
BEGIN failed--compilation aborted at /home/ekleog/.nix-profile/bin/dkim_responder.pl line 26.
```

As I don't really know the perl infrastructure in nixos (and not much more perl, actually), I don't know of any other way of fixing this than by going back to `buildPerlPackage`.

**Comment by Ekleog** [comment]
OK so it looks like I failed my test process and the issue is not the switch between `mkDerivation` and `buildPerlPackage` but rather an environment issue (running with `env -i` makes the failure appear).

I don't really know how I got to this, but will try pushing a corrected version as soon as possible.

**Comment by Ekleog** [comment]
Here it is, I guess my overriding the `configurePhase` removed the `preConfigure` from `pkgs/development/perl-modules/generic/builder.sh`; so here it is made "by hand".

Now `dkim_responder.pl` no longer fails when run under `env -i`.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**